changehc/tests/test_sensor.py

```diff
@@ -69,5 +69,5 @@ def test_fit_fips(self):
                 # binomial standard error, hence largest possible value is
                 # 100 * (0.5 / sqrt(MIN_DEN))
                 assert np.nanmax(res0["se"]) <= 100 * (0.5 / np.sqrt(Config.MIN_DEN))
-                assert np.nanmin(res0["se"]) > 0
+                assert np.nanmin(res0["se"]) >= 0
```

Comment:
Why is this necessary? With Jeffreys we shouldn't have se==0...

Reply:
What's Jeffreys?

For this case, sometimes rate = 0 (due to the explanation in the description), and then the se becomes 0 since its `rate(1-rate)`

Reply:
The Jeffreys prior for the binomial distribution adds a pseudosample to the data: half a count to each category, or +1 total. In theory it guarantees we never have a rate or se of 0.

It's applied here in the chng indicator:
https://github.com/cmu-delphi/covidcast-indicators/blob/85e1925c0d0c606651b303a09af6f7602e36037e/changehc/delphi_changehc/sensor.py#L123-L124

but it looks like that's done before smoothing. We should double-check with a statistician whether we should move it to after smoothing or not.

Reply:
who'd be best to consult on this with? 

If we move it to after smothing, isn't it possible we'd have a bunch of really low signals spike to 0.5/1 when they should be close to 0?

Reply:
Maria added the Jeffreys prior to the dv signal, where it was done before smoothing. I think we're getting significantly weirder behavior with the Change smoother, so maybe it's better to add it after smoothing to prevent the prior from getting smoothed out of existence. It's possible that we'd have low signals spike to 0.5/1 but we wouldn't be reporting those because the denominators would be way below threshold.

Reply:
I think it should be done *after* smoothing; doing it before smoothing is like adding half a count per day, instead of adding half a count per seven days, and hence biases the estimates more.

Since it's adding a half-count, its effect is small when the sample size is large. It should never change the estimates very much, unless you are reporting estimates when there are only two or three underlying data points.

Reply:
How would we add after smoothing? e.g. say it was originally 20/100=0.8, and it gets smoothed to 0.76 , would we do something like (76+0.5) / (100 + 1) = 0.757?

Reply:
Like a weighted average of (unsmoothed denominator) * (smoothed rate) and (1) * (0.5)? That seems reasonable, but also makes me wonder if that was the initial motivation behind smoothing the numerator and denominator separately. I'm not sure which approach is the most statistically correct, though I have a slight preference for doing the weighted average as you suggest, with the smoothed rate and the unsmoothed sample size.

Reply:
oh i didn't even think of the smooth vs unsmoothed denom -- I guess using 100 as the raw denom wasn't the most unambiguous choice. I was originally thinking of using the smoothed denom (so if it was 10/50 as the original ratio, you'd still get the same answer), though you're right that there's a few ways we could do this.

Reply:
> How would we add after smoothing? e.g. say it was originally 20/100=0.8, and it gets smoothed to 0.76 , would we do something like (76+0.5) / (100 + 1) = 0.757?

That's how the survey does it:
https://github.com/cmu-delphi/covidcast-indicators/blob/e7c7c13ef988d29e8d56409db8418a1de8a062ec/facebook/delphiFacebook/R/binary.R#L195-L206

Reply:
cool, lets go with that then.

Reply:
ok one slight hiccup. previously when we did it, we couldn't div/0 since the denominator would always be at least one. If we change to calculating the rate, smoothing, then adding the jeffreys at the end, we run the risk of getting nan values in the first step if denom = 0. Not sure how we want to deal with this, especially given @dshemetov's work on nan coding.

Reply:
Just to add on that last point: on my skim through the code, I didn't see any censorship in the sense of writing NANs when values are below a threshold. The backfill function seems to have a dual function of ensuring reasonable sample sizes and ensuring privacy by padding with time-averaging. So it seems to me that the only way to have a denominator of 0 would be to have a region with a week of 0 reports (which should be rare) or the same thing on the left boundary (more likely). Please correct me if I'm wrong on this mental model. @chinandrew could you point me to where the thresholding chucks the 0 denoms?